### PR TITLE
docs improvement in typos and redirection

### PR DESCRIPTION
**Pull request by ankatare**
Signed-off-by: Abhay Narayan Katare <abhay.katare@india.nec.com>

Commit Message: redirection issues in getting started docs for v3 API reference.
Additional Description: same has been identified in v1.15 and v1.16 as well ( see below urls)
https://www.envoyproxy.io/docs/envoy/v1.15.0/start/start
https://www.envoyproxy.io/docs/envoy/v1.16.0/start/start
Risk Level: LOW
Testing: unit
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
part of #11027

### CODE REVIEW

**Comment by ankatare** [comment]
in below url API reference is pointing to V2 API. please suggest if it is still ok, in my view it should point to v3 ?
https://www.envoyproxy.io/docs/envoy/v1.16.0/about_docs#about-the-documentation

**Comment by phlax** [comment]
i think this pr is re/based at against wrong branch, but otherwise lgtm

**Comment by ankatare** [comment]
@phlax sorry for this !!! would you please elaborate it, do you mean to push pr against 1.16/1/15  tag ?

File: docs/root/start/start.rst [hunk 1]
```diff
@@ -8,7 +8,7 @@ This section gets you started with a very simple configuration and provides some
 The fastest way to get started using Envoy is :ref:`installing pre-built binaries <install_binaries>`.
 You can also :ref:`build it <building>` from source.
 
-These examples use the :ref:`v3 Envoy API <envoy_api_reference>`, but use only the static configuration
+These examples use the :ref:`v3 Envoy API <envoy_v3_api_reference>`, but use only the static configuration
 feature of the API, which is most useful for simple requirements. For more complex requirements
```
Comment (phlax) [inline docs/root/start/start.rst:12]: this text is no longer present on `master`

the problem you are fixing is tho

**Comment by dio** [comment]
Sorry, do you want to fix docs for previous versions?

**Comment by ankatare** [comment]
@dio ahh.. i actually was reading docs and came across it. please suggest how to proceed ?

**Comment by dio** [comment]
You need to rebase your PR against https://github.com/envoyproxy/envoy/tree/release/v1.16 and https://github.com/envoyproxy/envoy/tree/release/v1.15 respectively if you want.

**Comment by ankatare** [comment]
@dio ok.

**Comment by ankatare** [comment]
@dio @phlax this issue still persist on master/main. 
se below url https://www.envoyproxy.io/docs/envoy/latest/api-v2/api#envoy-api-reference.

i believe this should fixed in master and all previous tagged releases ( 1.17, 1.16, 1.15 till we don't have EOL ? ) .
please suggest.

**Comment by alyssawilk** [comment]
let's do a main merge to get this fixed at HEAD, and then sort out docs releases.  SG?

**Comment by ankatare** [comment]
seems issues while rebase and merging with main. will recreate new PR.

**Comment by ankatare** [comment]
@alyssawilk sure. will do it in new PR.

**Pull request closed without merging**